### PR TITLE
Add a basic stage for converting dates to iso8601Z date (zulu date)

### PR DESCRIPTION
**Pull request by leonardsaers**
Solr consumes dates strictly formatted as iso8601 with time zone +0. The format is sometimes called zulu time. This stage formats dates into ISO8601 zulu format. So far it handles epoch time and ISO8601 with different time zones.

Is this stage worth adding to Hydras basic stages?

### CODE REVIEW

File: stages/processing/basic/src/main/java/com/findwise/hydra/stage/ZuluDateFormatter.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                
+                if ((contentField.getValue() instanceof String)) {
+                                
+                    parseDate(contentField, doc);
+                }
+            }
+        }
+    }
+
+    private void parseDate
```
Comment (laserval) [inline stages/processing/basic/src/main/java/com/findwise/hydra/stage/ZuluDateFormatter.java:86]: Weird formatting

**Comment by laserval** [comment]
I think a more generic implementation makes more sense to add. This is a good start for that, but it has too specific a name and purpose - why not simply `DateFormatterStage`? Then Zulu date can be a default, to start with.

**Comment by leonardsaers** [comment]
I agree that such a stage would be better. But it would add more complexity to the stage, converting from any format to any format.

To start with, should we just remain ZuluDateFormatter to DateFormatterStage?

File: stages/processing/basic/src/main/java/com/findwise/hydra/stage/ZuluDateFormatter.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    private String fromISO8601ToZuluDate(String date) {
+        DateTimeFormatter formatter = 
+                DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ssZ");
+        DateTime d = formatter.parseDateTime(date).withZone(DateTimeZone.UTC);
+        return d.toString(DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss'Z"));
+    }
+
+    private String fromEpocToZuluDate(String epoc) {
```
Comment (laserval) [inline stages/processing/basic/src/main/java/com/findwise/hydra/stage/ZuluDateFormatter.java:109]: Should be _epoch_

**Comment by laserval** [comment]
> To start with, should we just remain ZuluDateFormatter to DateFormatterStage?

Yes, that's a good start. The configuration doesn't look Zulu-specific, so it shouldn't be a problem to make the stage more generic in the future.

**Comment by leonardsaers** [comment]
Class name is now changed to DateFormatterStage

**Comment by laserval** [comment]
Closing in favor of merge of PR #308 , which is just a rebase of this to solve some merging issues.

**Pull request closed without merging**